src/components/VideoCard/VideoCard.js

```diff
@@ -1,25 +1,23 @@
 import React, { useState } from "react";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth, useWatchLater } from "../../context";
+import { addToWatchLater } from "../../services/addToWatchLater";
 import "./videoCard.css";
 
-export const VideoCard = ({
-	id,
-	title,
-	thumbnail,
-	channel,
-	views,
-	uploadedOn,
-}) => {
+export const VideoCard = ({ video }) => {
+	const { user } = useAuth();
+	const navigate = useNavigate();
+	const { watchLaterDispatch } = useWatchLater();
 	const [showEllipsis, setShowEllipsis] = useState(false);
 	return (
 		<>
 			<div className="single-video-card">
-				<Link to={`/watch/${id}`}>
-					<img src={thumbnail} alt={`img-${id}`} />
+				<Link to={`/watch/${video.id}`}>
+					<img src={video.thumbnail} alt={`img-${video.id}`} />
 				</Link>
 				<div className="video-name-and-menu">
-					<Link to={`/watch/${id}`}>
-						<h5 className="vid-title">{title}</h5>
+					<Link to={`/watch/${video.id}`}>
+						<h5 className="vid-title">{video.title}</h5>
```

Comment:
Improvement: You can de structure properties of video to avoid writing **video followed by property name**